addons/mail/models/mail_thread.py

```diff
@@ -1519,6 +1518,68 @@ def _mail_search_on_partner(self, normalized_emails, extra_domain=False):
             domain = expression.AND(domain, extra_domain)
         return self.env['res.partner'].search(domain)
 
+    @api.model
+    def _mail_find_user_for_gateway(self, emails, alias=None, records=None, fallback_model=False):
+        """ Utility method to find users from email addresses. Purpose is to link
+        document creation to users whenever possible, for example when creating
+        document through mailgateway.
+
+        :return list partners: a list of partner records ordered as given emails.
+          If no partner has been found and/or created for a given emails its
+          matching partner is an empty record.
+        """
+        if alias:
+            target_model = alias.alias_model_id.model
+        else:
+            target_model = fallback_model
+        if not target_model:
+            return [self.env['res.users'] for contact in emails]
+        fallback_user = self.env['res.users']
+        if alias and alias.alias_user_id:
+            try:
+                self.env[target_model].with_user(alias.alias_user_id).check_access_rights('create')
```

Comment:
if self.env[target_model].with_user(alias.alias_user_id).check_access_rights('create', raise_exception=False):
    fallback_user = alias.alias_user_id

-Cleaner
-Will skip some useless processing of error message